### PR TITLE
Add core version check

### PR DESCRIPTION
**Pull request by ludeeus**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast supervisor tests`)
- [x] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: supervisor/resolution/checks/core_version.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+class CheckCoreVersion(CheckBase):
+    """CheckCoreVersion class for check."""
+
+    async def run_check(self) -> None:
+        """Run check if not affected by issue."""
+        if not self.sys_homeassistant.version:
+            # Can't determine version.
+            return
+
+        if self.sys_homeassistant.version < "2021.1.3":
```
Comment (balloob) [inline supervisor/resolution/checks/core_version.py:28]: Does this need normalizing

Comment (ludeeus) [reply supervisor/resolution/checks/core_version.py:28]: It's handled by AwesomeVersion

**Review by balloob — APPROVED** [review]

File: supervisor/resolution/checks/core_version.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+"""Helpers to check core version."""
+from enum import Enum
+from pathlib import Path
+from typing import List
+
+from ...const import SUPERVISOR_DATA, CoreState
+from ..const import ContextType, IssueType
+from .base import CheckBase
+
+CC_PATH = Path(SUPERVISOR_DATA, "homeassistant", "custom_components")
```
Comment (pvizeli) [inline supervisor/resolution/checks/core_version.py:10]: use the `self.sys_config.path_homeassistant`

File: supervisor/resolution/checks/core_version.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            # Can't determine version.
+            return
+
+        if self.sys_homeassistant.version < "2021.1.3":
+            if CC_PATH.exists():
+                self.sys_resolution.create_issue(
+                    IssueType.VERSION,
+                    ContextType.CORE,
+                    reference=VersionReference.CUSTOM_COMPONENTS_BELOW_2021_1_3,
+                )
```
Comment (pvizeli) [inline supervisor/resolution/checks/core_version.py:34]: Suggestions=[SuggestionType.EXECUTE_UPDATE]

File: supervisor/resolution/const.py [hunk 4]
```diff
@@ -50,6 +50,7 @@ class IssueType(str, Enum):
     DOCKER_RATELIMIT = "docker_ratelimit"
     CORRUPT_DOCKER = "corrupt_docker"
     CORRUPT_REPOSITORY = "corrupt_repository"
+    VERSION = "version"
```
Comment (pvizeli) [inline supervisor/resolution/const.py:53]: Let's call it what it is -> SECURITY

File: supervisor/resolution/checks/core_version.py (hunk 1 above)
Comment (pvizeli) [reply supervisor/resolution/checks/core_version.py:28]: We need should catch the exception and skip the test if they get trow for version check. That makes sure the landing page i.e. don't break it, so we can save the check above

Comment (ludeeus) [reply supervisor/resolution/checks/core_version.py:28]: on landingpage you do not have any custom components yet

**Review by pvizeli — APPROVED** [review]

**Pull request merged**